### PR TITLE
MINOR: Various small improvements to kafka.metrics.MetricsTest

### PR DESCRIPTION
**Pull request by mimaison**
We've had to tweak the other tests to make them pass. 

`testBrokerTopicMetricsUnregisteredAfterDeletingTopic` seemed completely broken, as it's was creating a topic but producing/consuming to another one. 

With @mpburg

### CODE REVIEW

**Comment by ijuma** [comment]
The other tests don't require min.isr right? So, why are we setting at the class level?

**Comment by mimaison** [comment]
Yes it's true that they don't require it, so maybe let's merge your PR and I'll follow up with the improvements for the other test in another one

**Comment by ijuma** [comment]
Sounds good, thanks.

**Comment by ijuma** [comment]
I merged the other PR, so feel free to update this one.

File: core/src/test/scala/unit/kafka/metrics/MetricsTest.scala [hunk 1]
```diff
@@ -91,20 +92,20 @@ class MetricsTest extends KafkaServerTestHarness with Logging {
...
     sendMessages(servers, topic, nMessages)
     // create a consumer
     val consumerConfig1 = new ConsumerConfig(TestUtils.createConsumerProperties(zkConnect, group, consumerId))
     val zkConsumerConnector1 = new ZookeeperConsumerConnector(consumerConfig1, true)
-    val topicMessageStreams1 = zkConsumerConnector1.createMessageStreams(Map(topic -> 1), new StringDecoder(), new StringDecoder())
+    val topicMessageStreams1 = zkConsumerConnector1.createMessageStreams(Map(topic -> 1), new StringDecoder, new StringDecoder)
     getMessages(topicMessageStreams1, nMessages)
 
-    zkConsumerConnector1.shutdown()
+    zkConsumerConnector1.shutdown
```
Comment (ijuma) [inline core/src/test/scala/unit/kafka/metrics/MetricsTest.scala:108]: We include `()` for side-effecting methods. Will add this back before merging.

**Review by ijuma — APPROVED** [review]
Thanks for the PR, LGTM.

**Comment by ijuma** [comment]
Hmm, seems like `testBrokerTopicMetricsUnregisteredAfterDeletingTopic` fails transiently now:

https://jenkins.confluent.io/job/kafka-trunk/1937/testReport/junit/kafka.metrics/MetricsTest/testBrokerTopicMetricsUnregisteredAfterDeletingTopic/

**Comment by ijuma** [comment]
PR that makes assertion more informative: https://github.com/apache/kafka/pull/3042

**Pull request closed without merging**